MapleServer2/Types/Inventory.cs

```diff
@@ -132,15 +152,38 @@ public class Inventory {
             }
         }
 
+        // Checks if item exists already or if slot is taken already.
+        public bool Check(Item item)
+        {
+            if (items.ContainsKey(item.Uid)){
+                return true;
+            }
+            if (GetSlots(item.InventoryType).ContainsKey(item.Slot))
```

Comment:
Optional at this point but the items have max slot sizes, should we check this at this point?

Reply:
Probably. Will have to determine max slot size for items.

Reply:
can easily add that to the `Check` Method have it ran in the `Add` Method.